File: 프로그래머스/0/181924. 수열과 구간 쿼리 3/수열과 구간 쿼리 3.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = arr.clone();
        int temporary = 0;
        
        // ex) 01234 - 31204 - 32104 -  34102
        for(int i = 0; i < queries.length; i++){
            temporary = answer[queries[i][0]];
            answer[queries[i][0]] = answer[queries[i][1]];
            answer[queries[i][1]] = temporary;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181924. 수열과 구간 쿼리 3/수열과 구간 쿼리 3.py
import copy

def solution(arr, queries):
    answer = arr.copy()
    
    # 1)
#     for i in range(len(queries)):
#         a = queries[i][0]
#         b = queries[i][1]
        
#         temp = answer[a]
#         answer[a] = answer[b]
#         answer[b] = temp

    # 2) 다중 할당, 튜플 언패킹 -> 파이썬은 이게 된다 (임시변수 x)
    for a, b in queries:
        answer[a], answer[b] = answer[b], answer[a]
    
    return answer